### PR TITLE
test: cover MainActivity, Room migrations, preview providers, and Apps/Settings screens

### PR DESCRIPTION
**Pull request by Vladyslav-Soldatenko**
## Description

Raises unit-test coverage on the seven lowest-covered files identified by a JaCoCo report: `SettingsScreenKt`, `AppsListScreenKt`, `MainActivity`, `OverlayUiStateProvider`, `DojoUiStateProvider`, `RealTimeTicker`, and the Room `MIGRATION_*` objects.

## Type of Change

- [ ] New feature
- [ ] Bug fix
- [x] Refactoring (no functional changes)
- [ ] Documentation update
- [ ] Infrastructure / CI/CD
- [ ] Performance improvement
- [x] Tests

## Changes Made

- `MainActivity.kt`: extracted the inline onboarding `when` gate into a pure `resolveOnboardingStep()` function (behavior-preserving) and added `MainActivityTest` covering every step transition, including edge cases (skip flags, priority ordering).
- Added `RoomMigrationsTest.kt`, a JVM/Robolectric port of the existing instrumented `AppDatabaseMigrationTest` suite, covering `MIGRATION_1_2` through `MIGRATION_6_7` (schema changes, constraint enforcement, the position-renumbering and normalized-word-dedup logic) — this is what JaCoCo actually counts, since `jacocoTestReport` only runs `testDebugUnitTest`. Wired `app/schemas` into the `debug` source set's assets so `MigrationTestHelper` can find them under Robolectric (assets for unit tests come from the merged `debug` variant, not a test-specific merge).
- Relaxed `OverlayUiStateProvider`/`DojoUiStateProvider` (file-private preview providers) to `internal` and added direct tests asserting on their preview state sequences.
- Relaxed the `AppsListScreen`/`SettingsScreen` `@Preview` composables to `internal` and added smoke-render tests for them — this caught a pre-existing bug where `AppsListScreenErrorPreview` never set `isLoading = false`, so it rendered the loading spinner instead of the error text.
- Added `RealTimeTickerTest.kt` covering `now()` and the `nowTicks()` flow's timing (immediate first emission, 30s interval, non-completion).
- Extended the existing `SettingsContentTest.kt` with the `DialogState` branches it didn't yet cover (reviews-per-day, overlay interval, accessibility/export clicks, OpenAI API key/prompt/reverse-prompt dialogs, import row hidden when there are no import options, no-dialog-initially).

**Known, intentional gap:** `MainActivity`'s actual `onCreate`/Compose wiring and the top-level `hiltViewModel()` entry composables in `AppsListScreen`/`SettingsScreen` remain untested here — this repo has no Hilt-test harness (`hilt-android-testing`, `HiltTestApplication`), and building one from scratch is out of scope for a coverage PR. That wiring is already exercised end-to-end by the 15 existing instrumented e2e tests under `app/src/androidTest`.

## How to Test

1. `./gradlew testDebugUnitTest jacocoTestReport`
2. Open `app/build/reports/jacoco/jacocoTestReport/jacocoTestReport.xml` (or the HTML report) and confirm: `RealTimeTicker`, `OverlayUiStateProvider`, `DojoUiStateProvider` at 100%; `MigrationsKt$MIGRATION_*` at 94–100%; `AppsListScreenKt` ~78% (up from 45.9%); `SettingsScreenKt` ~68% (up from 58.7%).
3. `./gradlew ktlintCheck detekt lintDebug`

## Checklist

- [x] Code follows the project's style guidelines
- [x] Self-reviewed the code for obvious errors
- [x] Added or updated tests where applicable, including edge cases
- [x] Existing tests pass locally
- [ ] Updated documentation if needed
- [x] No duplication introduced. I searched the existing code for similar behavior and extracted similar parts to be reused
- [x] No new warnings or supressions introduced. If something like this has to be introduced - get approve from maintainer first (one `@Suppress("LongParameterList")` on the new pure `resolveOnboardingStep`, matching the existing suppression style already used elsewhere in `MainActivity.kt`)

---
_Generated by [Claude Code](https://claude.ai/code/session_011VXxZbZNLzb8hV1QKUi9U3)_

### CODE REVIEW

File: app/src/main/java/com/procrastilearn/app/MainActivity.kt [hunk 1]
```diff
@@ -45,6 +45,33 @@ internal val Context.dataStore by preferencesDataStore("app_prefs")
...
+internal enum class OnboardingStep {
+    LOADING,
+    ACCESSIBILITY,
+    OVERLAY,
+    LANGUAGE,
+    MAIN,
+}
+
+// Each parameter is an independent gate in the onboarding flow; grouping them into a holder
+// class would just move the same six flags one level down for no real benefit.
```
Comment (Vladyslav-Soldatenko) [inline app/src/main/java/com/procrastilearn/app/MainActivity.kt]: remove comment

File: app/src/test/java/com/procrastilearn/app/data/local/database/RoomMigrationsTest.kt [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+// JVM/Robolectric counterpart to the instrumented AppDatabaseMigrationTest so JaCoCo counts this
+// coverage; one test per migration/edge case for the same reason that file stays a single class.
+@Suppress("LargeClass")
+class RoomMigrationsTest {
```
Comment (Vladyslav-Soldatenko) [inline app/src/test/java/com/procrastilearn/app/data/local/database/RoomMigrationsTest.kt]: delete this file, let's not test it temporary

File: app/build.gradle.kts [hunk 3]
```diff
@@ -40,6 +40,10 @@ android {
       addAll(listOf("$projectDir/schemas", "src/androidTest/assets", "src/testShared/resources"))
     }
     getByName("test").resources.srcDirs("src/testShared/resources")
+    // Robolectric-based unit tests read assets from the merged "debug" variant assets (there is
+    // no separate merged-assets step for unit tests), so the schemas MigrationTestHelper needs
+    // must be added here rather than to the "test" source set.
+    getByName("debug").assets.srcDirs("$projectDir/schemas")
```
Comment (Vladyslav-Soldatenko) [inline app/build.gradle.kts]: let's try to revert it and not touch this file

File: app/src/main/java/com/procrastilearn/app/MainActivity.kt (hunk 1 above)
Comment (Vladyslav-Soldatenko) [reply app/src/main/java/com/procrastilearn/app/MainActivity.kt]: Done — removed.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: app/src/test/java/com/procrastilearn/app/data/local/database/RoomMigrationsTest.kt (hunk 2 above)
Comment (Vladyslav-Soldatenko) [reply app/src/test/java/com/procrastilearn/app/data/local/database/RoomMigrationsTest.kt]: Done — deleted `RoomMigrationsTest.kt`.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: app/build.gradle.kts (hunk 3 above)
Comment (Vladyslav-Soldatenko) [reply app/build.gradle.kts]: Done — reverted, `build.gradle.kts` is back to matching `master`.

---
_Generated by [Claude Code](https://claude.ai/code)_
